qbraid_qir/qasm3/visitor.py

```diff
@@ -1149,6 +1161,100 @@ def _visit_statement_block(block):
             one=lambda: _visit_statement_block(if_block),
         )
 
+    def _extract_values_from_discrete_set(self, discrete_set: DiscreteSet) -> List[int]:
+        """Extract the values from a discrete set.
+
+        Args:
+            discrete_set (DiscreteSet): The discrete set to extract values from.
+
+        Returns:
+            List[int]: The extracted values.
+        """
+        values = []
+        for value in discrete_set.values:
+            if not isinstance(value, IntegerLiteral):
+                self._print_err_location(discrete_set.span)
+                raise Qasm3ConversionError(
+                    f"Unsupported discrete set value {value} in discrete set"
+                )
+            values.append(value.value)
+        return values
+
+    def _visit_alias_statement(self, statement: AliasStatement) -> None:
+        """Visit an alias statement element.
+
+        Args:
+            statement (AliasStatement): The alias statement to visit.
+
+        Returns:
+            None
+        """
+        # pylint: disable=too-many-branches
+        target = statement.target
+        value = statement.value
+
+        alias_reg_name = target.name
```

Comment:
Hey, I don't see the update for this. Was it a parsing error, declaring the same alias again? If yes, then we can leave this out. Otherwise, it would be good to have a check here so that it does not conflict with other elements declared in the scope. 

I agree that the scope implementation has to be a bit more streamlined and we need to introduce "contexts" to correctly identify how far up we check in our scopes.